src/main/java/io/github/hypixel_api_wrapper/caching/CachingStrategy.java

```diff
@@ -0,0 +1,39 @@
+package io.github.hypixel_api_wrapper.caching;
+
+import io.github.hypixel_api_wrapper.util.Endpoint;
+import org.json.JSONObject;
+
+public interface CachingStrategy {
+
+    /**
+     * Saves the response of an endpoint to the cache
+     * @param endpoint the endpoint from where the response came from
+     * @param res the json response
+     */
+    void cacheResponse(Endpoint endpoint, JSONObject res);
+
+    /**
+     * Gets a response of the cache
+     * @param endpoint the endpoint where the object came from
+     * @return the response if one for the given endpoint was cached. {@code null} if not
+     */
+    JSONObject getCachedResponse(Endpoint endpoint);
```

Comment:
How would this method know what kind of data to look for? `Endpoint` is just an enum and doesn't have any info about the actual parameters used to fetch the data.

Reply:
Every response is stored by the endpoint it came from and as such is get form that endpoint.

Reply:
Sorry, I don't know what I meant by "kind". I mean, say you want to get the player named "BlackPlasma", how would the CachingStrategy know which player to check for if you're only giving it the endpoint?

Reply:
That's a good point. Should we use the URL then instead? Since it contains the query parameters or does hypixel have any post parameters?

Reply:
That's one way to do it, but we'd have to be extra careful about url consistency (ordering parameters the same, formatting uuids the same way, case-insensitivity, etc).

It also wouldn't allow us to query using different parameters than the original request. Like if I got a guild by its ID, but then the same guild was looked up by name, ideally it would still hit the cache since we have the name readily available.

Reply:
How do you suggest solving it then?